### PR TITLE
[MNG-7032] Evaluate --help and --version after configuring the logging/color

### PR DESCRIPTION
**Pull request by gnodet**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [ ] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

### CODE REVIEW

**Comment by gnodet** [comment]
Supersedes https://github.com/apache/maven/pull/415 with a simpler change that allows supporting `mvn -v -l file` too.

**Review by olamy — APPROVED** [review]

File: maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java [hunk 1]
```diff
@@ -423,7 +426,10 @@ void cli( CliRequest cliRequest )
             cliManager.displayHelp( System.out );
             throw e;
         }
+    }
 
+    private void command( CliRequest cliRequest ) throws ExitException
```
[the remark this answers is not in the document]
Comment (gnodet) [reply maven-embedder/src/main/java/org/apache/maven/cli/MavenCli.java]: I've rebased and renamed the method.

**Review by MartinKanters — APPROVED** [review]
Great! Compliments for the clarity of the unit test.

Running on [jenkins](https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven/job/MNG-7032/1/console)

**Comment by jira-importer** [comment]
Resolve #8005

**Pull request merged**